packages/plugin-core/src/test/suite-integ/SyncCommand.test.ts

```diff
@@ -67,4 +88,128 @@ suite("workspace sync command", function () {
+          // Create a new note so there are some changes
+          await NoteTestUtilsV4.createNote({
+            fname: "my-new-note",
+            body: "Lorem ipsum",
+            wsRoot,
+            vault: vaults[0],
+          });
+
+          const out = await new SyncCommand().execute();
+          const { committed, pulled, pushed } = out as any;
+          // The note added should get committed
+          expect(committed.length).toEqual(1);
+          // Should try to pull since allowed by configuration
+          expect(pulled.length).toEqual(1);
+          // Should not push since "noPush" is used
+          expect(pushed.length).toEqual(0);
+          done();
+        },
+      });
+    });
+
+    test("skip", (done) => {
+      runLegacyMultiWorkspaceTest({
+        ctx,
+        onInit: async ({ wsRoot, vaults }) => {
+          const remoteDir = tmpDir().name;
+          await GitTestUtils.createRepoForRemoteWorkspace(wsRoot, remoteDir);
+          await changeConfig(wsRoot, { workspaceVaultSync: "skip" });
+          // Create a new note so there are some changes
+          await NoteTestUtilsV4.createNote({
+            fname: "my-new-note",
+            body: "Lorem ipsum",
+            wsRoot,
+            vault: vaults[0],
+          });
+
+          const out = await new SyncCommand().execute();
+          const { committed, pulled, pushed } = out as any;
+          // Nothing should be done since "skip" is used
+          expect(committed.length).toEqual(0);
+          expect(pulled.length).toEqual(0);
+          expect(pushed.length).toEqual(0);
+          done();
+        },
+      });
+    });
+
+    test("sync", (done) => {
+      runLegacyMultiWorkspaceTest({
+        ctx,
+        onInit: async ({ wsRoot, vaults }) => {
+          const remoteDir = tmpDir().name;
+          await GitTestUtils.createRepoForRemoteWorkspace(wsRoot, remoteDir);
+          await changeConfig(wsRoot, { workspaceVaultSync: "sync" });
+          // Create a new note so there are some changes
+          await NoteTestUtilsV4.createNote({
+            fname: "my-new-note",
+            body: "Lorem ipsum",
+            wsRoot,
+            vault: vaults[0],
+          });
+
+          const out = await new SyncCommand().execute();
+          const { committed, pulled, pushed } = out as any;
+          // Should try doing everything since the config requires so
+          expect(committed.length).toEqual(1);
+          expect(pulled.length).toEqual(1);
+          expect(pushed.length).toEqual(1);
+          done();
+        },
+      });
+    });
+  });
 });
+
+async function changeConfig(
+  wsRoot: string,
+  overrideConfig: Partial<DendronConfig>
+) {
+  // Get old config, and override it with the new config
+  const serv = getWS().workspaceService!;
+  const config = serv.config;
+  await serv.setConfig(_.merge(config, overrideConfig));
+
+  // Commit this change, otherwise it will be a tracked file with changes which breaks git pull
```

Comment:
can you leave this comment at the function signature as well? this way it shows up in the IDE when people hover over the function signature